### PR TITLE
Custom buttons datepicker

### PR DESCRIPTION
**Pull request by Fenn-CS**
### What is the current behavior?

The calendar/time date picker used in various places on the site, does not have "Now"/"Infinity" button that can be easily used to set the time to those respective times.

Fixes: #3877 

### What is the new behavior?

Calendar now includes "Now" and "End of time" buttons.

now_and_end_of_time

### CODE REVIEW

**Comment by Fenn-CS** [comment]
Changes were made to `INSTALL_SUBMITTY_HELPER.sh` to test simply checkout the branch and run `sudo /usr/local/submitty/.setup/INSTALL_SUBMITTY.sh` in the VM

**Comment by bmcutler** [comment]
@Fenn-CS Please review all of the edits to this PR

It looks like you've made some changes to the forum twig (perhaps a bad merge with master or another branch).

Also it looks like you've made many unnecessary/cosmetic edits to some of the lines of code switching "<br />" to <br/>" which may not matter (?) to how they are rendered, but it's not a change to standard we are following in the project.  Making these changes makes it more likely we'll have conflicts with other code.

File: site/app/templates/course/CourseMaterials.twig [hunk 1]
```diff
@@ -56,96 +58,118 @@
...
+                            }
+                        ],
+                        label: "or",
+                        onClick: (index, fp) => {
+                            let date;
+                            switch (index) {
+                                case 0: date = new Date();
+                                    break;
+                                case 1: date = new Date("9999-01-01 00:00:00");
+                                    break;
```
Comment (MasterOdin) [inline site/app/templates/course/CourseMaterials.twig]: Put the line of code on the next line after the case line (indented).

File: site/app/templates/admin/admin_gradeable/AdminGradeableDates.twig [hunk 2]
```diff
@@ -123,12 +128,34 @@
...
+                            }
+                        ],
+                        label: "or",
+                        onClick: (index, fp) => {
+                            let date;
+                            switch (index) {
+                                case 0: date = new Date();
+                                    break;
+                                case 1: date = new Date("9999-01-01 00:00:00");
+                                    break;
```
Comment (MasterOdin) [inline site/app/templates/admin/admin_gradeable/AdminGradeableDates.twig]: put the executed line of code on the next line after the `case` line.

File: site/app/templates/admin/Extensions.twig [hunk 3]
```diff
@@ -49,30 +59,51 @@
...
+                            }
+                        ],
+                        label: "or",
+                        onClick: (index, fp) => {
+                            let date;
+                            switch (index) {
+                                case 0: date = new Date();
+                                    break;
+                                case 1: date = new Date("9999-01-01 00:00:00");
+                                    break;
```
Comment (MasterOdin) [inline site/app/templates/admin/Extensions.twig]: see comment below

**Comment by MasterOdin** [comment]
See @bmcutler note, and please fix merge conflicts. Looks to be fine other than some small tweaks.

File: site/package.json [hunk 4]
```diff
@@ -22,6 +22,7 @@
     "jquery-ui-timepicker-addon": "~1.6.3",
     "jquery.are-you-sure": "~1.9.0",
     "pdfjs-dist": "^2.0.943",
+    "shortcut-buttons-flatpickr": "^0.3.0",
```
[the remark this answers is not in the document]
Comment (Fenn-CS) [reply site/package.json]: @MasterOdin yes that was actually done. It's a freeze of packages that enabled me added it to `packages.json` and I think `INSTALL_SUBMITTY.sh` covers `npm install` using the updated `package.json` file. Or I am getting this wrong?

[the remark this answers is not in the document]
Comment (zjxiaohan) [reply site/package.json]: @Fenn-CS https://stackoverflow.com/questions/44206782/do-i-commit-the-package-lock-json-file-created-by-npm-5

[the remark this answers is not in the document]
Comment (MasterOdin) [reply site/package.json]: Yeah, that link is a good read on it.

The gist of it is that if you update the contents of either package.json or composer.json, you should be running the `install` command, and then commit the updated lock file along with the the updated json file. Pins the version, sets a hash to verify what we're downloading, etc.

[the remark this answers is not in the document]
Comment (Fenn-CS) [reply site/package.json]: Ok thanks, would address this and mark it as so when it's done.

[the remark this answers is not in the document]
Comment (Fenn-CS) [reply site/package.json]: Done

File: site/app/templates/course/CourseMaterials.twig (hunk 1 above)
Comment (Fenn-CS) [reply site/app/templates/course/CourseMaterials.twig]: Done

**Comment by Fenn-CS** [comment]
@bmcutler  changes like `<br />` to `<br/>` are appearing because @thef1stofhername merged masted into this branch this introduced a couple of file changes which I did not actually make myself.

**Comment by thef1stofhername** [comment]
Yeah sorry about the spacing stuff I was trying to help with fixing the merge conflict

**Comment by Fenn-CS** [comment]
No issues @thef1stofhername you and perhaps @zjxiaohan, @MasterOdin can review the issue once more!
I am not sure if merging master was the right thing but given that there were merged conflicts, I guess this has saved some time for the merger.

**Comment by bmcutler** [comment]
@Fenn-CS It shows that there are still merge conflicts, and that this PR is changing 75 files.
(Looks like the recent merged PRs are tangled up.)
Something is definitely not right.   

Which files do you expect to change with this PR?  
It might be simplest to start with a fresh branch from master and reapply only the desired edits.

**Comment by bmcutler** [comment]
Also, I see the edits to the br tag spaces in @Fenn-CS 's first commit to this branch.
I don't think those were first introduced by @thef1stofhername

**Pull request closed without merging**